manta-trusted-setup/src/groth16/ceremony/util.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019-2022 Manta Network.
+// This file is part of manta-rs.
+//
+// manta-rs is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// manta-rs is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with manta-rs.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Utilities
+
+extern crate alloc;
+
+use manta_crypto::arkworks::pairing::Pairing;
+use manta_util::{
+    serde::{de::DeserializeOwned, Serialize},
+    BoxArray,
+};
+use std::{
+    fs::File,
+    io::{Read, Write},
+    path::Path,
+};
+
+use crate::groth16::mpc::State;
+
+use super::message::ServerSize;
+
+/// Logs `data` to a disk file at `path`.
+#[inline]
+pub fn log_to_file<T, P>(path: &P, data: &T)
+where
+    T: Serialize,
+    P: AsRef<Path>,
+{
+    let mut file = File::create(path).expect("Open file should succeed.");
```

Comment:
consider others for `File::create`. May pass configuration (e.g., [OpenOption](https://doc.rust-lang.org/nightly/std/fs/struct.OpenOptions.html)) as a parameter.